src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/CountOperation.java

```diff
@@ -20,15 +20,32 @@
  * Operation that returns count of documents based on the filter condition. Written with the
  * assumption that all variables to be indexed.
  */
-public record CountOperation(CommandContext commandContext, LogicalExpression logicalExpression)
+public record CountOperation(
+    CommandContext commandContext, LogicalExpression logicalExpression, int pageSize, int limit)
     implements ReadOperation {
 
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
     SimpleStatement simpleStatement = buildSelectQuery();
-    return countDocuments(queryExecutor, simpleStatement)
+    Uni<CountResponse> countResponse = null;
+    if (limit == -1) countResponse = countDocuments(queryExecutor, simpleStatement);
+    else countResponse = countDocumentsByKey(queryExecutor, simpleStatement);
+
+    return countResponse
         .onItem()
-        .transform(docs -> new CountOperationPage(docs.count()));
+        .transform(
+            docs -> {
+              if (limit == -1) {
+                return new CountOperationPage(docs.count(), false);
+              } else {
+                boolean moreData = docs.count() > limit();
+                return new CountOperationPage(
+                    docs.count() == 0
+                        ? 0
```

Comment:
Is there need to separately check for `0`? Seems like that'd work without extra check.

Reply:
Good Catch, fixed it.